### ISSUE TITLE
Appending rows with empty cells results in increasing row indentation

### ISSUE THREAD

**Issue by hiliev** [burnash/gspread#537]
### Environment info

Operating System: Windows 10
Python version: 3.6.3 (Anaconda)
gspread version: both 2.0.0 and 2.1.1

### Steps to reproduce
```python
ws = sheet.worksheet('Sheet1')
ws.append_row(['Test1', '', 'Test2'])
ws.append_row(['Test1', '', 'Test2'])
ws.append_row(['Test1', '', 'Test2'])
```

### Output

![image](https://user-images.githubusercontent.com/5229193/38493065-22061470-3bf1-11e8-9eb0-5253aac6445c.png)

It looks like new rows appended with `append_row()` start from the last populated column on the previous rows. This is a relatively new behaviour that started a day or so ago.

**Comment by burnash** [comment]
@hiliev thanks for reporting this. Can you clarify that it used to work correctly with gspread version >=2.0.0 before?

**Comment by hiliev** [comment]
Yes, it was working with version 2.0.0 before.

I only upgraded to 2.1.1 today. Actually, about a week ago the new rows started appearing indented, all with the same indent, i.e. started from column N instead of A. At first I thought that someone has messed up the spreadsheet with hidden columns or filters. Today the rows started appearing with increasing indent until the number of cells limit got hit (similar to #516) and I upgraded to 2.1.1 to see whether it fixes the problem.

**Comment by audrummer15** [comment]
+1 on this

**Comment by ZhanaraDev** [comment]
Was it fixed?

**Comment by audrummer15** [comment]
No this is still an issue as of 3.0.2 I believe

**Comment by venkateshkotte** [comment]
Is it fixed?

**Comment by JeffJassky** [comment]
... Still happening.

**Comment by burnash** [comment]
It looks like this is not a bug. I tried to run `append_row()` with different table layouts and it appears that the row shift happens when the Sheets API has multiple logical "tables" to chose from and it selects the rightmost. It's explained in details in the example in the Sheets API guide here: https://developers.google.com/sheets/api/guides/values#appending_values

So In this particular case, when appending the second row the Sheets API sees two logical "tables" A1 and C1 and decides to append to C1:

![example](https://user-images.githubusercontent.com/264674/74592253-c060d080-501f-11ea-9449-f953e9f0deb9.png)

When appending the third the same happens with E2 and so on.

As the Sheets API guide suggests the solution is to explicitly tell the Sheets API which table it needs to append to. For that, we'll need to pass an explicit `range` parameter to the API call. I've implemented this in #726 (and named the parameter `table_range`).

So for this particular case, after #726 is merged you'll need to call `append_row` this way:

```python
ws.append_row(['Test1', '', 'Test2'], table_range='A1')
```
That way the API will know that you'd like to append to the table that starts with A1 cell.
Again, this is *only* need in the case where there's ambiguity in the selection of logical "tables".
